packages/account/src/Components/poi/poi-form-on-signup/idv-doc-submit-on-signup/idv-doc-submit-on-signup.tsx

```diff
@@ -13,17 +18,12 @@ import {
     validate,
     validateName,
 } from '../../../../Helpers/utils';
-import FormSubHeader from '../../../form-sub-header';
-import IDVForm from '../../../forms/idv-form';
-import PersonalDetailsForm from '../../../forms/personal-details-form.jsx';
-import FormFooter from '../../../form-footer';
 
 type TIdvDocSubmitOnSignup = {
     citizen_data: FormikValues;
     onPrevious: (values: FormikValues) => void;
     onNext: (values: FormikValues, action: FormikHelpers<FormikValues>) => void;
     value: FormikValues;
-    has_idv_error?: boolean;
     account_settings: GetSettings;
     getChangeableFields: () => string[];
     residence_list: ResidenceList;
```

Comment:
```suggestion
    residence_list: NonNullable<ReturnType<typeof useResidenceList>['data']>;
```
why this method instead of just getting the types from api-types 
Incase the data is we passing as props is undefined and we get type error, we can do it like this
`Exclude<NonNullable<ReturnType<typeof useResidenceList>['data'][0]['text']>, undefined>;`
This will make sure that text from residence list is not undefined and we dont have to do like {text || 'fallback'}

Reply:
`ResidenceList` is imported  from `@deriv/api-types`. When the type is available why do we need to infer it from a hook's return value?

Reply:
The Exclude approach is nice one. I'll remember it where I know the value will not be undefined. Thanks

Reply:
Because if we want to transform the data for example which is not in the api-types. This will be an issue.